cpp/benchmarks/io/json/nested_json.cpp

```diff
@@ -24,15 +24,95 @@
 
 #include <tests/io/fst/common.hpp>
 
+#include <cudf/detail/utilities/vector_factories.hpp>
 #include <cudf/scalar/scalar_factories.hpp>
 #include <cudf/strings/repeat_strings.hpp>
 #include <cudf/types.hpp>
 
-#include <cstdlib>
+#include <string>
+#include <vector>
 
-namespace cudf {
 namespace {
-auto make_test_json_data(size_type string_size, rmm::cuda_stream_view stream)
+
+// List of List nested.
+std::string generate_list_of_lists(int32_t max_depth, int32_t max_rows, std::string elem)
```

Comment:
AFAIK, appending strings like this is very slow, since you keep making copies of the old data every (or at least often) time you append another string. I recommend using [stringstream](https://en.cppreference.com/w/cpp/io/basic_stringstream) instead.
All functions except `generate_json` should modify passed `stringstream` and then finally `generate_json` can return a string using `stringstream_object.str()`.

Reply:
Tried passing `string&` after `reserve` and use `append`. It's slower than simply using `+=`. (2^30 string_size took 1m23s compared to 1m16s).
I profiled the code and found that `std::to_string`, `operator+`, string ctor took most time in decreasing order.
I will make some adjustments to reduce benchmark data generation time.

Reply:
Wait, you found that the code was slower after reserving than before? Something sounds fishy there. I wouldn't expect meaningful differences between `+=` and `append`, but maybe there are tiny error checking/exception handling overheads in one but not the other. Preallocating should definitely be faster though unless something is very wrong.

In any case, not a huge deal for this use case. It sounds as if you were able to reduce benchmark generation time, which is nice and I'm happy with that result however you did it.

Reply:
Very interesting finds. Maybe not critical for this code, but good to know in general :)
I can't rationalize why preallocation would be slower.
All in all, no concerns with data gen time in this PR :)

Reply:
Top level string pre-allocation helps reduce a few seconds. But passing that string as `string&` to all functions and using `append`/`+=` was slower. I am not sure what might be the issue. (I didn't try this again after adding row cache).